File: EEX-4465-Data Structures and Algorithms/Java/Sorting-methods/BubbleSort.java
// File: BubbleSort.java

public class BubbleSort {

    static void bubbleSort(int[] arr) {
        int n = arr.length;
        boolean swapped;
        for (int i = 0; i < n - 1; i++) {
            swapped = false;

            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // Swap
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;

                    swapped = true;
                }
            }

            // If no two elements were swapped, break
            if (!swapped)
                break;
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 25, 12, 22, 11};
        bubbleSort(arr);
        System.out.print("Sorted array: ");
        for (int num : arr)
            System.out.print(num + " ");
    }
}

File: EEX-4465-Data Structures and Algorithms/Final-Exam-Works/Pseudo-codes/bubbleSort.py
# Pseudocode: BubbleSort(Array)
FUNCTION BubbleSort(Arr)
    SET n = Arr.length
    FOR i = 0 TO n-2
        FOR j = 0 TO n-2-i
            IF Arr[j] > Arr[j+1] THEN
                SWAP Arr[j] and Arr[j+1]
            ENDIF
        ENDFOR
    ENDFOR
END FUNCTION


This Pseudocode is for the Bubble Sort algorithm, a method to sort an array.

Here's how it works 

    1. Outer Loop: Controls the number of passes over the array. After each pass, the largest 
    remaining unsorted element moves to its correct final position.

    2. Inner Loop: Compares each pair of adjacent elements.

    3. Swap: If an element is larger than the one next to it, they swap positions.


This Process repeats until the entire array is sorted from smallest to largest.
